### PR TITLE
esp32: Add a gdb script to show backtrace

### PR DESCRIPTION
**Pull request by yamt**
## Summary
Just for my convenience. Hopefully useful for others too.

### CODE REVIEW

**Comment by Ouss4** [comment]
@yamt for reference I have this little script that I use with the backtrace.  It requires a few manual input though (copying the backtrace to a file).
```shell
for bt in `cat $1 | cut -d':' -f3`; do
  xtensa-esp32-elf-addr2line -pfiaCs -e nuttx $bt
done
```

**Comment by Ouss4** [comment]
BTW, did you try this script with an interrupt in the way?
For the builtin backtrace support, interrupts need a fabricated (pseudo) save area to emulate a call4.

File: tools/esp32/backtrace.gdbscript [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    set $a1 = $arg2
+    set $pc_topbits = (int)$pc & 0xc0000000
+    # The return address from xtensa_sig_deliver to _xtensa_sig_trampoline
+    set $sig_tramp_ra = (int)(_xtensa_sig_trampoline + 2 * 3)
+    print/a $pc
+    while (1)
+        # Note: "- 3" to workaround the case where "call" is
+        # the last instruction in the calling function.
+        set $next_pc = (($a0 & 0x3fffffff) | $pc_topbits) - 3
+        print/a $next_pc
```
Comment (Ouss4) [inline tools/esp32/backtrace.gdbscript:48]: This just prints the backtrace.  What's different from the backtrace already dumped?

Comment (yamt) [reply tools/esp32/backtrace.gdbscript:48]: do you mean CONFIG_XTENSA_DUMPBT_ON_ASSERT?

* CONFIG_XTENSA_DUMPBT_ON_ASSERT dumps the fault handler stack but not user's
* this version deals with signal trampoline
* ability to specify pc/a0/a1 is handy for me

**Comment by yamt** [comment]
> @yamt for reference I have this little script that I use with the backtrace. It requires a few manual input though (copying the backtrace to a file).
> 
> ```shell
> for bt in `cat $1 | cut -d':' -f3`; do
>   xtensa-esp32-elf-addr2line -pfiaCs -e nuttx $bt
> done
> ```

btw, can you consider to put the script somewhere in the tree? (regardless of this PR)

**Comment by yamt** [comment]
> BTW, did you try this script with an interrupt in the way?
> For the builtin backtrace support, interrupts need a fabricated (pseudo) save area to emulate a call4.

no. my motivation at this point was user stack.

**Review by Ouss4 — APPROVED** [review]

**Comment by Ouss4** [comment]
> > @yamt for reference I have this little script that I use with the backtrace. It requires a few manual input though (copying the backtrace to a file).
> > ```shell
> > for bt in `cat $1 | cut -d':' -f3`; do
> >   xtensa-esp32-elf-addr2line -pfiaCs -e nuttx $bt
> > done
> > ```
> 
> btw, can you consider to put the script somewhere in the tree? (regardless of this PR)

It's #2679. Please take a look.

**Pull request merged**